packages/amplication-data-service-generator/src/update-package-jsons.ts

```diff
@@ -12,64 +12,27 @@ import { merge } from "lodash";
 export function updatePackageJSONs(
   modules: Module[],
   baseDirectory: string,
-  update: Record<string, any>
+  update: { [key: string]: any }[]
 ): Module[] {
   return modules.map((module) => {
-    if (module.path === `${baseDirectory}/package.json`) {
-      return updatePackageJSON(module, update);
-    } else if (module.path === `${baseDirectory}/package-lock.json`) {
-      return updatePackageLockJSON(module, update);
-    }
-    return module;
-  });
-}
-
-function updatePackageJSON(module: Module, update: Record<string, any>) {
-  let pkg = JSON.parse(module.code);
-  pkg = merge(pkg, update);
-
-  if (!semver.valid(pkg.version)) {
-    delete pkg.version;
-  }
+    const updateModule =
+      module.path === `${baseDirectory}/package.json`
+        ? updatePackageJSON(module, update)
+        : module;
 
-  return {
-    ...module,
-    code: JSON.stringify(pkg, null, 2),
-  };
+    return updateModule;
+  });
 }
 
-/**
- *
- *
- * @param update object that contain the update data to enter instead of the default data as name and version
- * @returns the updated module
- *
- * In npm 7 and higher the package lock contains a packages section (lockfile v2)
- * to see the npm docs of this update https://github.blog/2021-02-02-npm-7-is-now-generally-available/#changes-to-the-lockfile
- */
-function updatePackageLockJSON(
-  module: Module,
-  update: Record<string, any>
-): Module {
-  const lockfile = JSON.parse(module.code);
-  /**
-   * The v2 lockfile contains a package in the packages section that is a mirror of the project so we need to update the static data there also
-   */
-  const pkg = lockfile.packages[""];
-
-  Object.assign(lockfile, update);
-  Object.assign(pkg, update);
-
-  if (!semver.valid(lockfile.version)) {
-    delete lockfile.version;
-  }
+function updatePackageJSON(module: Module, update: { [key: string]: any }[]) {
+  const pkg = preparePackageJsonFile(module, update);
 
   if (!semver.valid(pkg.version)) {
     delete pkg.version;
   }
 
   return {
     ...module,
-    code: JSON.stringify(lockfile, null, 4),
+    code: pkg,
```

Comment:
what is the difference between pkg and JSON.stringify(pkg, null, 2) ?

Reply:
It isn't, but pkg returns it so I use it instead